lib/SILOptimizer/FunctionSignatureTransforms/ExistentialTransform.cpp

```diff
@@ -192,13 +189,11 @@ void ExistentialSpecializerCloner::cloneAndPopulateFunction() {
   /// If there is an argument with no DestroyUse, insert DeallocStack
   /// before return Instruction.
   llvm::SmallPtrSet<ReturnInst *, 4> ReturnInsts;
-  if (MissingDestroyUse) {
-    /// Find the set of return instructions in a function.
-    for (auto &BB : NewF) {
-      TermInst *TI = BB.getTerminator();
-      if (auto *RI = dyn_cast<ReturnInst>(TI)) {
-        ReturnInsts.insert(RI);
-      }
+  /// Find the set of return instructions in a function.
+  for (auto &BB : NewF) {
```

Comment:
This fixes the old code that was inserting dealloc_stack after destroy_addr which is wrong. dealloc_stack are now inserted just before return instructions. See test case: $s7dealloc12wrap_foo_ncp1a1bSiAA1P_pz_AaE_pztFTf4ee_n below.